src/Hadolint/Rules.hs

```diff
@@ -182,18 +194,14 @@ commentMetadata :: ShellCheck.Interface.Comment -> Metadata
 commentMetadata (ShellCheck.Interface.Comment severity code message) =
     Metadata (Text.pack ("SC" ++ show code)) severity (Text.pack message)
 
-shellcheckBash :: ParsedFile -> [RuleCheck]
-shellcheckBash = concatMap check
-  where
-    check (InstructionPos (Run args) source linenumber) =
-        argumentsRule (applyRule source linenumber) args
-    check _ = []
-    applyRule source linenumber args =
-        rmDup [RuleCheck m source linenumber False | m <- convert args]
-    convert args = [commentMetadata c | c <- Bash.shellcheck args]
-    rmDup :: [RuleCheck] -> [RuleCheck]
-    rmDup [] = []
-    rmDup (x:xs) = x : rmDup (filter (\y -> metadata x /= metadata y) xs)
+shellcheckBash :: Rule
+shellcheckBash = mapInstructions checkBash ()
+  where
+    checkBash :: CheckerWithState ()
+    checkBash st _ (Run (ArgumentsList script)) = (st, doCheck script)
```

Comment:
I plan to use the state variable `st` later to record the type of `SHELL` to parse